Makefile

```diff
@@ -501,6 +501,11 @@ fmt: tools ## Run goimports
 vet: ## Run go vet
 	$(GO) vet ./...
 
+check: misspell lint
+
+misspell:
+	hack/check/check-misspell.sh
```

Comment:
nit: I wonder if we should instead rename the script `misspell.sh` as its already under `check` dir.